### PR TITLE
feat(adapter): add cooldown period for tee set up

### PR DESCRIPTION
**Pull request by envin3**
Cooldown period for new tee set

### CODE REVIEW

File: cpp/contracts/pam.hpp [hunk 1]
```diff
@@ -38,12 +38,17 @@ namespace eosio {
...
+            public_key updating_key;
             bytes attestation;
+            bytes updating_attestation;
+            uint64_t change_grace_threshold = 0;
         };
 
         using tee_pubkey = singleton<"tee"_n, tee>;
         typedef eosio::multi_index<"mappings"_n, mappings> mappings_table;
 
+        static constexpr uint64_t TEE_ADDRESS_CHANGE_GRACE_PERIOD = 172800;
```
Comment (gitmp01) [inline cpp/contracts/pam.hpp:50]:
```suggestion
        static constexpr uint64_t TEE_ADDRESS_CHANGE_GRACE_PERIOD = 172800; // 48 hours
```

**Review by gitmp01 — CHANGES_REQUESTED** [review]
Looks good overall, could you add a test showing the rejection when someone tries to update the pubkey before the grace period? You can control time in tests through:
```js
const { TimePointSec } = require('@wharfkit/antelope')
const timestamp = TimePointSec.fromMilliseconds(Date.now())

blockchain.setTime(timestamp)
```

**Review by gitmp01 — APPROVED** [review]

File: cpp/test/adapter.test.js [hunk 2]
```diff
@@ -402,6 +405,36 @@ describe('Adapter tests', () => {
...
+
+      await expectToThrow(action, errors.GRACE_PERIOD_NOT_ELAPSED)
+    })
+
+    it('Should be able to apply the new tee key after the grace period', async () => {
+      const timestamp = TimePointSec.fromMilliseconds(
+        Date.now() + TEE_ADDRESS_CHANGE_GRACE_PERIOD_MS,
+      )
+
+      blockchain.setTime(timestamp)
```
Comment (envin3) [inline cpp/test/adapter.test.js:422]: nice

**Review by envin3 — COMMENTED** [review]
LGTM

**Pull request merged**